src/tests/pages/index.test.js

```diff
@@ -37,8 +37,10 @@ describe("App", () => {
     it("displays the titles from course search results", () => {
         render(<App isSuccessfulSearch searchResults={searchResults} searchTerm="foobar" />);
 
-        expect(screen.getByRole("link", { name: "Maths" })).toBeInTheDocument();
-        expect(screen.getByRole("link", { name: "Physics" })).toBeInTheDocument();
+        const headings = screen.getAllByRole("heading");
+
+        expect(headings[2]).toHaveTextContent("Maths");
+        expect(headings[3]).toHaveTextContent("Physics");
```

Comment:
I believe there was an unresolved discussion about magic numbers in the previous PR. I think the risk here is that if we add new components and/or headings before the rendering of results, this test will fail - for me ideally that wouldn't be the case, as we would still be meeting the acceptance criteria in the test's title.

I think `CourseSearchResults.test.js` is the right place to test the ordering (which you've done!), I'd suggest keeping it more generic here, which also removes that prerequisite knowledge that there are (currently) two headings before the results:

```suggestion
        expect(screen.getByRole("heading", { name: "Maths" })).toBeInTheDocument();
        expect(screen.getByRole("heading", { name: "Physics" })).toBeInTheDocument();
```

I don't think it's a huge deal, but it's nice to avoid issues in future if we spot them now :slightly_smiling_face: